pkg/cloudprovider/providers/gce/gce_loadbalancer_internal.go

```diff
@@ -139,6 +155,15 @@ func (gce *GCECloud) ensureInternalLoadBalancer(clusterName, clusterID string, s
 		return nil, err
 	}
 
+	// The addrMgr does not exist if no targetIP was specified. This function was not deferred
+	// because the code should continue to hold the targetIP if the sync fails for any reason.
+	// At this point, the existing rule is known to exist; therefore, releasing the address is fine.
+	if addrMgr != nil {
```

Comment:
Did not read the entire ILB file, just want to ask whether you need to `defer` the `ReleaseAddress` to reduce the time the controller holds the static IP. Perhaps the IP quota is not a concern for internal IPs?

Reply:
I'm confused, we discussed this case offline. If the service sync fails for any reason, I thought you said we should hold onto the IP for the next sync, right?

Reply:
We should hold on to the IP if the forwarding rule is not using it. If the forwarding rule IS using the IP, it's safe to release the it since we can always get it back. This is what the external LB do today for optimization. If there is no need for ILB to do so, we don't have to add more complexity.